servers/rendering/rendering_native_surface_external_target.cpp

```diff
@@ -0,0 +1,442 @@
+/**************************************************************************/
+/*  rendering_native_surface_external_target.cpp                          */
```

Comment:
Can you provide some context to why this exists?

Right off the bat, its a huge red flag to have so much OpenGL code leaking out of the driver folder. 

I am very wary of having an entire system that appears to duplicate what already exists, but sitting in an entirely different part of the code base. I suspect that this design (whatever its intended purpose) will be an enormous maintenance burden and will basically result in LibGodot being extremely brittle and prone to breaking from seemingly unrelated changes